File: app/src/main/java/by/drawgrid/library/model/Dot.java
package by.drawgrid.library.model;


import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.Rect;
import android.graphics.Typeface;

import by.drawgrid.library.view.ViewPlusGrid;


public class Dot extends Element {

	Point point;
	
	 public int radiusCircle=2;
	float circleWidht=4;
	Paint paintCircle;
	Paint paintText;

	
	public Dot(){
		super(Constant.TYPE_DOT);
		point = new Point();
		init(ViewPlusGrid.DPI_DENSITY);
	}
	
	
	public Dot(Point p){
		super(Constant.TYPE_DOT);
		point = p;
		init(ViewPlusGrid.DPI_DENSITY);
	}
	
	
	
	
	
	void init( float dpi) {
		circleWidht *= dpi;
		radiusCircle *= dpi;
		paintCircle = new Paint();
		paintText = new Paint();
		
		paintCircle.setStyle(Paint.Style.FILL_AND_STROKE);
		paintCircle.setColor(Color.parseColor("#33CC99"));
		paintCircle.setStrokeWidth(circleWidht);
		
		paintText.setStyle(Paint.Style.FILL_AND_STROKE);
		paintText.setColor(Color.BLACK);
		paintText.setTypeface(Typeface.DEFAULT_BOLD);
		paintText.setTextSize(20);
	}
	
	public void draw(Canvas canvas) {
		
		
		if( pointInDrawRect(point)){
		Point pointDraw =  ViewPlusGrid.calcDrawCoordinate(point);
//		float x = ViewPlusGrid.DrawKoeficientX*(point.x-ViewPlusGrid.pointMIN.x);
//		float y = ViewPlusGrid.H-ViewPlusGrid.DrawKoeficientY*(point.y-ViewPlusGrid.pointMIN.y);
		float x = pointDraw.x;
		float y = pointDraw.y;
		
		 
		canvas.drawCircle(x, y , radiusCircle, paintCircle);

		StringBuilder strText = new StringBuilder();
		strText.append(point.y);
		String label = strText.toString();

		Paint paint = new Paint();
		Rect bounds = new Rect();
		paint.getTextBounds(label, 0, label.length(), bounds);

		bounds.width();

		canvas.drawText(label, x- (bounds.width() / 2), y- (radiusCircle + 3 + bounds.height()), paintText);
		}
	}
	
	public static boolean pointInDrawRect(Point point){
		 if((point.x>ViewPlusGrid.decartPointEnd.x)||(point.y>ViewPlusGrid.decartPointEnd.y)||(point.x < ViewPlusGrid.decartPointStart.x)||(point.y < ViewPlusGrid.decartPointStart.y)){
			 return false;
		 }
		
		return true;
	}

	@Override
	public int getColor() {
		// TODO Auto-generated method stub
		return 0;
	}

	@Override
	public void setColor(int color) {
		// TODO Auto-generated method stub
		
	}

	@Override
	public float getWidthLine() {
		// TODO Auto-generated method stub
		return 0;
	}


	@Override
	public void setWidthLine(float widthLine) {
		// TODO Auto-generated method stub
		
	}
	

}

File: app/src/main/java/by/drawgrid/library/model/kotlin/Dot.kt
package by.drawgrid.library.model.kotlin

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.graphics.Rect
import by.drawgrid.library.model.Constant
import by.drawgrid.library.model.Point
import by.drawgrid.library.view.kotlin.GridCalculator


class Dot : BaseElement {

    internal var point: Point

    var radiusCircle = 5
    internal var circleWidht = 4f
    lateinit var paintCircle: Paint
    lateinit var paintText: Paint

    constructor(context: Context?) : super(Constant.TYPE_DOT, context) {
        point = Point()
        init(dpi)
    }


    internal fun init(dpi: Float) {
        circleWidht *= dpi
        radiusCircle *= dpi.toInt()
        paintCircle = Paint()
        paintText = Paint()

        paintCircle.style = Paint.Style.FILL_AND_STROKE
        paintCircle.color = Color.GREEN
        paintCircle.strokeWidth = circleWidht

        paintText.style = Paint.Style.STROKE
        paintText.color = Color.BLACK
        paintText.textSize = 16f
    }

    override fun draw(canvas: Canvas) {


        if (pointInDrawRect(point, gridCalculator!!)) {
            val pointDraw = gridCalculator?.calcDrawCoordinate(point)
            //		float x = ViewPlusGrid.DrawKoeficientX*(point.x-ViewPlusGrid.pointMIN.x);
            //		float y = ViewPlusGrid.H-ViewPlusGrid.DrawKoeficientY*(point.y-ViewPlusGrid.pointMIN.y);
            val x = pointDraw?.x
            val y = pointDraw?.y


            canvas.drawCircle(x!!, y!!, radiusCircle.toFloat(), paintCircle)

            val strText = StringBuilder()
            strText.append(point.x).append(" , ").append(point.y)
            val label = strText.toString()

            val paint = Paint()
            val bounds = Rect()
            paint.getTextBounds(label, 0, label.length, bounds)

            bounds.width()

            canvas.drawText(label, x - bounds.width() / 2, y - (radiusCircle + 3 + bounds.height()), paintText)
        }
    }

    fun pointInDrawRect(point: Point, gridCalculator: GridCalculator): Boolean {
        return if (point.x > gridCalculator.pointMAX!!.x || point.y > gridCalculator.pointMAX!!.y || point.x < gridCalculator.pointMIN!!.x || point.y < gridCalculator.pointMIN!!.y) {
            false
        } else true

    }

    override fun getColor(): Int {
        // TODO Auto-generated method stub
        return 0
    }

    override fun setColor(color: Int) {
        // TODO Auto-generated method stub

    }

    override fun getWidthLine(): Float {
        // TODO Auto-generated method stub
        return 0f
    }


    override fun setWidthLine(widthLine: Float) {
        // TODO Auto-generated method stub

    }
}